pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -320,12 +328,12 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error
 			}
 			tr.Status.SetCondition(&apis.Condition{
 				Type:    apis.ConditionSucceeded,
-				Status:  corev1.ConditionFalse,
+				Status:  succeededStatus,
 				Reason:  reason,
 				Message: fmt.Sprintf("%s: %v", msg, err),
 			})
 			c.Recorder.Eventf(tr, corev1.EventTypeWarning, "BuildCreationFailed", "Failed to create build pod %q: %v", tr.Name, err)
-			c.Logger.Errorf("Failed to create build pod for task %q :%v", err, tr.Name)
+			c.Logger.Errorf("Failed to create build pod for task %q: %v", tr.Name, err)
```

Comment:
what do you think about moving the logic around determining the status of the Run out of this function and into a separate function that could have its own tests?